From c768ed307d08484cd61ad8ac3faac330d22c8775 Mon Sep 17 00:00:00 2001
From: Luc Maisonobe <luc@apache.org>
Date: Wed, 8 Apr 2015 17:36:24 +0200
Subject: [PATCH] Fixed wrong handling of a corner case.

The problem occurs in very difficult to reproduce cases, as it is really
a corner case. When the vanishing cut-subhyperplane occurs during a
Difference operation, the fixed node still has a null cut-subhyperplane.
This was not properply handled.

JIRA: MATH-1162
---
 .../apache/commons/math4/geometry/partitioning/BSPTree.java    | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/main/java/org/apache/commons/math4/geometry/partitioning/BSPTree.java b/src/main/java/org/apache/commons/math4/geometry/partitioning/BSPTree.java
index 73d3133667..3be5b084f1 100644
--- a/src/main/java/org/apache/commons/math4/geometry/partitioning/BSPTree.java
+++ b/src/main/java/org/apache/commons/math4/geometry/partitioning/BSPTree.java
@@ -670,6 +670,9 @@ public void insertInTree(final BSPTree<S> parentTree, final boolean isPlusChild,
                     plus      = fixed.plus;
                     minus     = fixed.minus;
                     attribute = fixed.attribute;
+                    if (cut == null) {
+                        break;
+                    }
                 }
 
             }
